collab-document/src/document.rs

```diff
@@ -1,44 +1,225 @@
+    });
+
+    document
+  }
+
+  pub fn to_json(&self) -> Result<serde_json::value::Value, DocumentError> {
+    let txn = self.root.transact();
+    let root_id = self.root.get(&txn, "head_id").unwrap().to_string(&txn);
+    let document_data = serde_json::json!({
+        "document": {
+            "root_id": root_id,
+            "blocks": self.blocks.to_json(),
+            "meta": serde_json::json!({
+                "text_map": self.text_map.to_json(),
+                "children_map": self.children_map.to_json(),
+            })
+        }
+    });
+
+    Ok(document_data)
+  }
+
+  pub fn init(&self, txn: &mut TransactionMut) {
+    let head_id = self.root.get(txn, "head_id").unwrap().to_string(txn);
+    let head_children_id = Uuid::new_v4().to_string();
+    let head_text_id = Uuid::new_v4().to_string();
+    let head_data = BlockData {
+      text: head_text_id,
+      level: None,
+    };
+    // { document: { blocks: { head_id: { id: "head_id", ty: "page", data: { text: "head_text_id", level: null }, children: "head_children_id" } } } }
+    self.insert_block(
+      txn,
+      head_id.clone(),
+      "page".to_string(),
+      head_data,
+      head_children_id,
+      "".to_string(),
+      "".to_string(),
+    );
+
+    let first_id = Uuid::new_v4().to_string();
+    let first_text_id = Uuid::new_v4().to_string();
+    let first_children_id = Uuid::new_v4().to_string();
+    let first_data = BlockData {
+      text: first_text_id,
+      level: None,
+    };
+    // { document: { blocks: { head_id: { id: "head_id", ty: "page", data: { text: "head_text_id", level: null }, children: "head_children_id" }, first_id: { id: "first_id", ty: "text", data: { text: "first_text_id", level: null }, children: "first_children_id" } } } }
+    self.insert_block(
+      txn,
+      first_id,
+      "text".to_string(),
+      first_data,
+      first_children_id,
+      head_id,
+      "".to_string(),
+    );
+  }
+
+  pub fn insert_block(
+    &self,
+    txn: &mut TransactionMut,
+    block_id: String,
+    ty: String,
+    data: BlockData,
+    children_id: String,
+    parent_id: String,
+    prev_id: String,
+  ) {
+    self
+      .children_map
+      .create_children_with_txn(txn, children_id.clone());
+    self.text_map.create_text(txn, data.text.as_str());
+
+    let block = self
+      .blocks
+      .create_block(txn, block_id, ty, parent_id, children_id.clone(), data);
+    let block = match block {
+      Ok(block) => block,
+      Err(err) => {
+        panic!("create block error: {:?}", err);
+      },
+    };
+    self.insert_block_to_parent(txn, &block, prev_id);
+  }
+
+  pub fn insert_block_to_parent(&self, txn: &mut TransactionMut, block: &Block, prev_id: String) {
+    let parent_id = &block.parent;
+    if parent_id == "" {
+      return;
     }
+    let parent = self.blocks.get_block(txn, parent_id);
+    let parent = match parent {
+      Some(parent) => parent,
+      None => {
+        panic!("parent not found");
```

Comment:
You can return a Result<(), DocumentError> 

You can extend the DocumentError.